provenance-verifier/pkg/workflow.go

```diff
@@ -119,6 +122,23 @@ func (w *Workflow) validateJobSteps(job *actionlint.Job) error {
 	return nil
 }
 
+// =============== Re-usable workflow ================ //
+func (w *Workflow) IsJobCallingTrustedReusableWorkflow(job *actionlint.Job) (bool, error) {
+	if job == nil || job.WorkflowCall == nil || job.WorkflowCall.Uses == nil {
+		return false, nil
+	}
+
+	values := strings.Split(job.WorkflowCall.Uses.Value, "@")
```

Comment:
should we include that the error may be that it's not pinned (assuming that if no tag is mentioned it defaults to main)

Reply:
ill add a better message.
Do we want to enforce pinning by hash? Currently it accepts pinning by version (easier for debugging), but we may want to make this a hash.
I'll create a tracking issue for this.